public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -0,0 +1,195 @@
+import { AccessControlAction } from 'app/types';
+
+import { AddCorrelationForm } from './Forms/AddCorrelationForm';
+import { EditCorrelationForm } from './Forms/EditCorrelationForm';
+import { EmptyCorrelationsCTA } from './components/EmptyCorrelationsCTA';
+import { Column, Table } from './components/Table';
+import { CorrelationData, useCorrelations } from './useCorrelations';
+
+const sortDatasource: SortByFn<CorrelationData> = (a, b, column) =>
+  a.values[column].name.localeCompare(b.values[column].name);
+
+const isSourceReadOnly = ({ source }: Pick<CorrelationData, 'source'>) => source.readOnly;
+
+const loaderWrapper = css`
+  display: flex;
+  justify-content: center;
+`;
+
+export default function CorrelationsPage() {
+  const navModel = useNavModel('correlations');
+  const [isAdding, setIsAdding] = useState(false);
+  const { correlations, create, remove, update, isLoading, error } = useCorrelations();
+  const canWriteCorrelations = contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
+
+  const handleAdd = useCallback<ComponentProps<typeof AddCorrelationForm>['onSubmit']>(
+    async (correlation) => {
+      await create(correlation);
+      setIsAdding(false);
+    },
+    [create]
+  );
+
+  const RowActions = useCallback(
+    ({
+      row: {
+        original: {
+          source: { uid: sourceUID, readOnly },
+          uid,
+        },
+      },
+    }: CellProps<CorrelationData, void>) =>
+      !readOnly && (
+        <DeleteButton aria-label="delete correlation" onConfirm={() => remove({ sourceUID, uid })} closeOnConfirm />
+      ),
+    [remove]
+  );
+
+  const columns = useMemo<Array<Column<CorrelationData>>>(
+    () => [
+      {
+        cell: InfoCell,
+        shrink: true,
+        visible: (data) => data.some(isSourceReadOnly),
+      },
+      {
+        id: 'source',
+        header: 'Source',
+        cell: DataSourceCell,
+        sortType: sortDatasource,
+      },
+      {
+        id: 'target',
+        header: 'Target',
+        cell: DataSourceCell,
+        sortType: sortDatasource,
+      },
+      { id: 'label', header: 'Label', sortType: 'alphanumeric' },
+      {
+        cell: RowActions,
+        shrink: true,
+        visible: (data) => canWriteCorrelations && data.some(negate(isSourceReadOnly)),
+      },
+    ],
+    [RowActions, canWriteCorrelations]
+  );
+
+  const data = useMemo(() => correlations, [correlations]);
+
+  const showEmptyListCTA = data?.length === 0 && !isAdding && (!error || error.status === 404);
+
+  const showErrorAlert = error && error.status !== 404;
+
+  return (
+    <Page navModel={navModel}>
+      <Page.Contents>
+        <div>
+          <HorizontalGroup justify="space-between">
+            <div>
+              <h4>Correlations</h4>
+              <p>Define how data living in different data sources relates to each other.</p>
+            </div>
+            {canWriteCorrelations && data?.length !== 0 && data !== undefined && !isAdding && (
```

Comment:
It doesn't seem to work - the button is still visible after clicking on it:


https://user-images.githubusercontent.com/745532/186107837-954e19dd-4111-454b-9edc-c5993b69f62e.mov

Reply:
this does a different thing, it removes the "add new", not the submit button

Reply:
(working on it though, hold on :D )